kof-operator/webapp/collector/tests/victoria_tests/VictoriaDetailsPage.test.tsx

```diff
@@ -0,0 +1,113 @@
+import "@testing-library/jest-dom";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import { fakeVictoriaResponse } from "../fake_data/fake_victoria_response";
+import { act, render, renderHook, screen } from "@testing-library/react";
+import VictoriaDetailsPage from "../../src/components/pages/victoriaPage/victoria-details/VictoriaDetailsPage";
+import { useVictoriaMetricsState } from "../../src/providers/victoria_metrics/VictoriaMetricsProvider";
+import { MetricsDatabase } from "../../src/database/MetricsDatabase";
+
+describe("Victoria List", () => {
+  beforeEach(() => {
+    vi.stubGlobal(
+      "fetch",
+      vi.fn().mockResolvedValue({
+        ok: true,
+        json: () => Promise.resolve(fakeVictoriaResponse),
+      })
+    );
+
+    vi.spyOn(MetricsDatabase.prototype, "getRecords").mockResolvedValue([]);
+    vi.spyOn(MetricsDatabase.prototype, "addRecord").mockResolvedValue();
+    vi.spyOn(MetricsDatabase.prototype, "deleteOldRecords").mockResolvedValue();
+    vi.spyOn(MetricsDatabase.prototype, "deleteRecord").mockResolvedValue();
+
+    vi.mock("react-router-dom", async () => {
+      const actual = await vi.importActual("react-router-dom");
+      return {
+        ...actual,
+        useNavigate: () => vi.fn(),
+      };
+    });
+  });
+
+  afterEach(() => {
+    vi.resetAllMocks();
+    useVictoriaMetricsState.setState({
+      isLoading: false,
+      data: null,
+      selectedCluster: null,
+      selectedPod: null,
+      error: undefined,
+    });
+  });
+
+  it("should render victoria details page", async () => {
+    const { result } = renderHook(() => useVictoriaMetricsState());
+    await result.current.fetch();
+
+    await act(async () => {
+      result.current.setSelectedCluster("aws-ue2");
+    });
+
+    await act(async () => {
+      result.current.setSelectedPod(
+        result.current.selectedCluster?.pods[0].name || ""
+      );
+    });
+
+    const selectedPod = result.current.selectedPod;
+
+    render(<VictoriaDetailsPage />);
+
+    expect(
+      screen.getByText(`VictoriaLogs Select: ${selectedPod.name}`)
+    ).toBeInTheDocument();
+
+    const tablist = screen.getByRole("tablist");
+    expect(tablist).toBeInTheDocument();
+    expect(tablist.querySelectorAll("button")).toHaveLength(5);
+
+    expect(screen.getByText("Overview")).toBeInTheDocument();
+    expect(screen.getByText("System")).toBeInTheDocument();
+    expect(screen.getByText("Go Runtime")).toBeInTheDocument();
+    expect(screen.getByText("Network")).toBeInTheDocument();
+    expect(screen.getByText("VictoriaLogs Select")).toBeInTheDocument();
+  });
+
+  it("should show loading spinner when data is loading", () => {
+    vi.stubGlobal(
+      "fetch",
+      vi.fn().mockResolvedValue({
+        ok: true,
+        json: () => {
+          new Promise(function (resolve) {
+            setTimeout(resolve, 5000);
```

Comment:
Please use fake timers to avoid waiting 5 sec in test
https://vitest.dev/guide/mocking.html#timers